frontend/src/metabase/visualizations/click-actions/actions/HideColumnAction/HideColumnAction.tsx

```diff
@@ -48,10 +47,9 @@ export const HideColumnAction: LegacyDrill = ({
           };
         }
 
-        return onUpdateVisualizationSettings({
-          "table.columns": columnSettingsCopy,
-        });
+        return question.updateSettings({ "table.columns": columnSettingsCopy });
       },
+      questionChangeBehavior: "updateQuestion",
```

Comment:
Default behaviour of the `question` callback is to change and run the card. We don't want to run the question every time we hide the column though, so we add the `updateQuestion` behaviour to simply update the question.